learning_resources/utils.py

```diff
@@ -493,3 +493,74 @@ def offeror_delete_actions(offeror: LearningResourceOfferor):
     pm = get_plugin_manager()
     hook = pm.hook
     hook.offeror_delete(offeror=offeror)
+
+
+def _walk_ocw_topic_map(
+    topics: dict, parent: None | LearningResourceTopic = None
+) -> None:
+    """
+    Walk the topic map provided and create topic records accordingly.
+
+    This will recursively walk through the topics list and create/update topic
+    records as appropriate. There's just names here so if the record exists with
+    the same name (and parent), it'll update; otherwise, it creates.
+
+    Args:
+    - topics (dict): the topics to process
+    - parent (None or LearningResourceTopic): the parent topic (for inner loops)
+    Returns:
+    - None
+    """
+
+    for topic in topics:
+        lr_topic, _ = LearningResourceTopic.objects.filter(
```

Comment:
I think this is where the duplicate topic gets created since the get_or_create includes the parent

Reply:
Yes - this should have used an update_or_create rather than the get_or_create. As is the topics weren't getting saved so the channel topic detail records were also getting created incorrectly. Fixed in https://github.com/mitodl/mit-open/pull/844/commits/be4c4184ef6763891705530a8b2a23304081ef90 .